File: ERP_Anass_app/src/app/models/Sale/Sale.ts
export interface Sale {
  idSale?: number;
  saleRef?: string;
  totalAmount?: number;
  TotalPayment?: number;
  idCurrency?: number;
  paymentStatus?: string;
  paymentDate?: Date;
  saleDate?: Date;
  createdAt?: Date;
  updatedAt?: Date;
  remarks?: string;
  idCustomer?: number;
  isActive?: boolean;
  currencyName?: string;
  customerName?: string;

  // Additional Fields
  saleStatus?: string; // e.g., "Pending", "Approved", "Received"
  expectedDeliveryDate?: Date;
  actualDeliveryDate?: Date;
  shippingAddress?: string;
  saleType?: string; // e.g., "Raw Material", "Services"
  discountAmount?: number;
  discountPercentage?: number;
  taxRate?: number;
  totalTaxAmount?: number;
  exchangeRate?: number;
  approvedBy?: string;
  approvalDate?: Date;
  paymentTerms?: string; // e.g., "Net 30", "Net 60"
  saleChannel?: string; // e.g., "Online", "Offline"
}

File: ERP_SERVER/ERP_Anass_backend/Models/Sale.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
namespace ERP_Anass_backend.Models
{
    public class Sale
    {

        [Key]
        public int IdSale { get; set; }
        public string SaleRef { get; set; }

        public decimal? TotalAmount { get; set; }
        public decimal? TotalPayment { get; set; }
        public string PaymentStatus { get; set; }
        public DateTime? PaymentDate { get; set; }
        public DateTime? SaleDate { get; set; }
        public DateTime? CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public string? Remarks { get; set; }
        public bool? IsActive { get; set; }

        public int? IdCurrency { get; set; }
        [JsonIgnore]
        public Currency? Currencyobj { get; set; }

        public int? IdCustomer { get; set; }
        [JsonIgnore]
        public Customer? Customerobj { get; set; }
        public string SaleStatus { get; set; }
        public DateTime? ExpectedDeliveryDate { get; set; }
        public DateTime? ActualDeliveryDate { get; set; }
        public string? ShippingAddress { get; set; }
        public string? SaleType { get; set; }
        public decimal? DiscountAmount { get; set; }
        public decimal? DiscountPercentage { get; set; }
        public decimal? TaxRate { get; set; }
        public decimal? TotalTaxAmount { get; set; }
        public decimal? ShippingAmount { get; set; }
        public decimal? ExchangeRate { get; set; }
        public string? ApprovedBy { get; set; }
        public DateTime? ApprovalDate { get; set; }
        public string? PaymentTerms { get; set; }
        public string? SaleChannel { get; set; }

        [JsonIgnore]
        public ICollection<SaleDetails> SaleDetails { get; set; } = [];
    }
}
